staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -245,11 +284,16 @@ func (w *deferredResponseWriter) Write(p []byte) (n int, err error) {
 	return w.w.Write(p)
 }
 
-func (w *deferredResponseWriter) Close() error {
+func (w *deferredResponseWriter) Close() (err error) {
 	if !w.hasWritten {
-		return nil
+		if !w.hasBuffered {
+			return nil
+		}
+		_, err := w.unbufferedWrite(w.buffer)
```

Comment:
add a comment here explaining that we never reached the gzip writer minimum size, so we can return without needing to do the gzip writer cleanup